### PR TITLE
New function: MPI_Comm_create_group

### PR DESCRIPTION
**Pull request by variemai**
Since there is no reference of MPI_Comm_create_group  I decided to make one! Please check if this makes sense to you. My intention is to compare with MPI_Comm_create that is why I used  the same example.  I have run the examples in my machine and they work.

### CODE REVIEW

**Review by rookiehpc — CHANGES_REQUESTED** [review]
**Remark**: you will see there is a high number of comments, however by no means does this mean the pull request is of poor quality. Instead, it exhibits the details, guidelines and advice that are missing from the contribution guide. As pull requests come in, the contribution guide will grow and hopefully become more and more clear, concise and complete. Given that this is the very first pull request, it greatly helps calibrate the contribution guide so on top of actually contributing to the RookieHPC project, it also helps improving the contribution process itself, so **thank you**. (Reviewing your pull request even helped notice mistakes from the `MPI_Comm_create` page)

The above being now clear, you will see that a certain number of comments are minor syntactic tweaks such as a missing full dot or extraneous space or underscore.

In addition, you will see that, apart from the two comments below, comments are put directly with a suggested updated code, so the modification process is likely to be very quick. Regarding the two comments that could not be put directly in the code:
- Remove the binary file `a.out`.
- Run the python script `build_lookup.py`. This will update the sitemap, integrate `MPI_Comm_create_group` in the search bar, in the index pages etc...

If you have any question, please feel free to ask. You can also reply directly to comments if needed :)

File: mpi/docs/mpi_comm_create_group/data.json [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+                },
+                {
+                    "Name":"group",
+                    "Description":"The group containing the processes from the old_communicator to include in the new communicator. It can contain the entire group of processes of the old communicator, or just a subgroup or MPI_GROUP_EMPTY.",
+                    "Type":"INTEGER",
+                    "Optional":false
+                },
+                {
+                    "Name":"tag",
+                    "Description":"If multiple threads at a given process perform concurrent MPI_Comm_Create_Group operations, the user must distinguish these operations by providing different tag or comm arguments",
```
[the remark this answers is not in the document]
Comment (variemai) [reply mpi/docs/mpi_comm_create_group/data.json:103]: I certainly agree with this but I thought this remark was plain and I did not want to say a lot more. I also found the "wildcard" term philosophical, that is why I chose not to write it.

File: mpi/docs/mpi_comm_create_group/data.json [hunk 2]
```diff
@@ -88,25 +88,25 @@
...
                     "Name":"group",
-                    "Description":"The group containing the processes from the old_communicator to include in the new communicator. It can contain the entire group of processes of the old communicator, or just a subgroup or MPI_GROUP_EMPTY.",
+                    "Description":"The group containing the processes from the old communicator to include in the new communicator. It can contain the entire group of processes of the old communicator, or a subgroup or MPI_GROUP_EMPTY.",
                     "Type":"INTEGER",
                     "Optional":false
                 },
                 {
                     "Name":"tag",
-                    "Description":"If multiple threads at a given process perform concurrent MPI_Comm_Create_Group operations, the user must distinguish these operations by providing different tag or comm arguments",
+                    "Description":"If multiple threads at a given process perform concurrent MPI_Comm_Create_Group operations, the user must distinguish these operations by providing different tag or comm arguments. Tag cannot be MPI_ANY_TAG.",
```
Comment (rookiehpc) [inline mpi/docs/mpi_comm_create_group/data.json:103]: Copy description from C or FORTRAN-2008 versions.

**Review by rookiehpc — CHANGES_REQUESTED** [review]

**Review by rookiehpc — APPROVED** [review]

**Comment by rookiehpc** [comment]
**Congratulations, you have officially become the first contributor to the RookieHPC project!** 🎉

**Pull request merged**